pipeline/git/git_test.go

```diff
@@ -173,3 +173,19 @@ func TestSnapshotDirty(t *testing.T) {
 	ctx.Snapshot = true
 	assert.NoError(t, Pipe{}.Run(ctx))
 }
+
+func TestShortCommitHash(t *testing.T) {
+	_, back := testlib.Mktmp(t)
+	defer back()
+	testlib.GitInit(t)
+	testlib.GitCommit(t, "first")
+	var ctx = context.New(config.Project{
+		Snapshot: config.Snapshot{
+			NameTemplate: "{{.Commit}}",
+		},
+	})
+	ctx.Snapshot = true
+	ctx.Config.Git.ShortHash = true
+	assert.NoError(t, Pipe{}.Run(ctx))
+	assert.Regexp(t, `^[a-z0-9]{7}$`, ctx.Version)
```

Comment:
would probably be easier to use `assert.Len`

Reply:
First I thought so too, but then there is no guarantee that `{{ .Commit }}` really contains a Git hash or something else that just happens to be 7 characters long.

But I guess that would be another test, to check that we picked up an actual hash and not something else from the CLI when calling Git. For this specific feature though, `assert.Len` should be enough.

I will change it later.

Reply:
makes sense indeed, thanks!